.github/workflows/build.yml

```diff
@@ -0,0 +1,27 @@
+name: build-go-projects
+
+on:
+  merge_group:
+  push:
+    branches: [main]
+  pull_request:
+    branches: ["*"]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version: '1.22'
+          cache: false
+      - name: Checkout code
```

Comment:
The first "uses" step checksout, do we need to do it again?

Reply:
Probably not